src/exchanges/files.ts

```diff
@@ -0,0 +1,61 @@
+import { print } from 'graphql';
+import { extractFiles } from 'extract-files';
+import { filter, pipe, map } from 'wonka';
+import { Operation } from 'src/types';
+
+const fileExchange = ({ forward }) => {
+  const isMutation = (operation: Operation) =>
+    operation.operationName === 'mutation';
+
+  return ops$ => {
+    const preFlight$ = pipe(
+      ops$,
+      filter(isMutation),
+      map(operation => {
+        const { url } = operation.context;
+        let { files } = extractFiles(operation);
+        if (!files.length) return true;
```

Comment:
Since it's a map I check for the length

Reply:
When you map the operation, shouldn't you return the operation here instead of returning a boolean?